controller/search.py

```diff
@@ -30,11 +51,55 @@ def get(self):
             neighbors = []
             for k, v in answer.neighbors().items():
                 neighbors.append(str(v))
+            meta = {
+                "point": (args['x'], args['y']),
+                "epsg": args['epsg']
+            }
             return {
-                        "cell_id": str(answer),
+                        "meta": meta,
+                        "dggs_cell_id": str(answer),
                         "sub_cells": sub_cells,
                         "neighbors": neighbors
                     }
         except exceptions.CRSError:
             return api.abort(500, message='Please input a valid epsg code')
 
+resolutionParser = reqparse.RequestParser()
+resolutionParser.add_argument('resolution', type=int, required=True, choices=[1,2,3,4,5,6,7,8,9,10,11,12,13,14], help='DGGS Resolution 1 to 14')
+
+@api.route('/find_dggs_for_a_line')
+@api.doc(parser=resolutionParser)
+class FindDGGSForALine(Resource):
+    @api.expect(linestring,  validate=True)
+    def post(self):
+        # no find dggs for a line function found at DGGS Engine
+        pass
+
+def bbox(coord_list):
+     box = []
+     for i in (0,1):
+         res = sorted(coord_list, key=lambda x:x[i])
+         box.append((res[0][i],res[-1][i]))
+     ret = [ box[0][0], box[1][0], box[0][1], box[1][1] ]
+     return ret
+@api.route('/find_dggs_cells_within_polygon')
+@api.doc(parser=resolutionParser)
+class FindDGGSForALine(Resource):
+    @api.expect(polygon,  validate=True)
+    def post(self):
+        args = resolutionParser.parse_args()
+        polygon = shape(request.json)
```

Comment:
@jevy-wangfei should this input accept any geojson feature as input? i.e. Polygon, Multipolygon, GeometryCollection, Feature, FeatureCollection ?
e.g. This example doesn't work https://github.com/CSIRO-enviro-informatics/AusPIX_DGGS/blob/geojson/test_data/ACT_SA1_Black_Mountain.geojson

Reply:
Only single polygon could pass the validation as defined here: https://github.com/CSIRO-enviro-informatics/AusPIX_DGGS_API/pull/10/files#diff-00f4989b2d24ee061aeb0c7c98b465edR13
Should we support muti-polygon?